### PR TITLE
chore: upgrade dependencies

### PR DESCRIPTION
**Pull request by czosel**

### CODE REVIEW

File: src/pragma.mjs [hunk 1]
```diff
@@ -1,4 +1,4 @@
-import memoize from "mem";
+import memoize from "memoize";
```
Comment (fisker) [inline src/pragma.mjs:1]: Maybe we can replace it with this?

```js
const memoize = getPageLevelDocBlock => {
	const cache = new Map();
	return text => {
		if (!cache.has(text)) {
			cache.set(text, getPageLevelDocBlock(text));
		}
		return cache.get(text);
	};
};
```

Comment (czosel) [reply src/pragma.mjs]: Thanks for the review @fisker! I put a generic memoize function in the utils and dropped the dependency. Shall we still drop support for Node 16?

Comment (fisker) [reply src/pragma.mjs]: I don't know, personally I don't use old versions, but Prettier still support Node.js v14, maybe worth to keep support for 16 if not cost much to maintenance.

Comment (fisker) [reply src/pragma.mjs]: Looks like we don't have `engine` field at all, does that mean we can drop them anytime :smile:

Comment (czosel) [reply src/pragma.mjs]: :+1: I reverted the change to the node versions in CI. Where do I see the v14 baseline in prettier? `engines` is set to `>=18`: https://github.com/prettier/prettier/blob/main/package.json#L24

Comment (fisker) [reply src/pragma.mjs]: `engines` in https://unpkg.com/browse/prettier@3.2.4/package.json

Comment (fisker) [reply src/pragma.mjs]: https://github.com/prettier/prettier/blob/8cbee2e217baad7acf4cb3947834e8c1b41ed647/scripts/build/build-package-json.js#L40

Comment (czosel) [reply src/pragma.mjs]: Ah I see, thanks :) shall we add engines with node >= 16 here?

Comment (fisker) [reply src/pragma.mjs]: I don't know, I never use `engines` myself.

**Review by fisker — APPROVED** [review]

File: src/pragma.mjs (hunk 1 above)
Comment (czosel) [reply src/pragma.mjs]: Then I’d say let’s leave it as is, ok?

**Pull request merged**